### PR TITLE
PVRTC texture compression support

### PR DESCRIPTION
**Pull request by gleblebedev**
PvrTexTool can produce a dds file. With this change rbfx/Urho3D can read the files and properly display them. Sample project is attached.

### CODE REVIEW

**Comment by gleblebedev** [comment]
PvrTexTool available at the link: https://www.imaginationtech.com/developers/powervr-sdk-tools/pvrtextool/

File: Source/Urho3D/Graphics/OpenGL/OGLTexture.cpp [hunk 1]
```diff
@@ -187,6 +187,31 @@ bool Texture::IsCompressed() const
...
+            unsigned blocksWide = Max(2, width / xSize);
+            unsigned blocksHeight = Max(2, height / 4);
+            return blocksWide * blocksHeight * blockSize;
+        }
+        return GetRowDataSize(width) * ((height + 3u) >> 2u);
+    }
+    else
+        return GetRowDataSize(width) * height;
+}
+
```
Comment (rokups) [inline Source/Urho3D/Graphics/OpenGL/OGLTexture.cpp:214]: I am not entirely sure if moving `GetDataSize()` to subclasses is a good idea. In theory compression formats do not depend on graphics API, they depend on hardware. For example even though OpenGL can use PVRTC on iOS - this format is not available on androids. Also consider (do not laugh!) we add a metal backend. Both OpenGL and metal could use PVRTC so this code would have to be duplicated. So i think its best to have this logic in `Texture.cpp`. @eugeneko any thoughts?

Comment (gleblebedev) [reply Source/Urho3D/Graphics/OpenGL/OGLTexture.cpp:214]: Format seems to be platform dependent. Would be nice if the texture format field would be not int but enum...

Comment (rokups) [reply Source/Urho3D/Graphics/OpenGL/OGLTexture.cpp:214]: Format is platform dependent, but code attached it to graphics API, not a platform.

Comment (gleblebedev) [reply Source/Urho3D/Graphics/OpenGL/OGLTexture.cpp:214]: I mean format depends on graphics Api. It is exposed as int but the actual meaning depends on API you use.

Comment (rokups) [reply Source/Urho3D/Graphics/OpenGL/OGLTexture.cpp:214]: Not really. Both metal and opengl can use pvrtc, just like opengl and directx can use DXT5. Compressed formats depend on hardware, not graphics API.

**Pull request closed without merging**